### PR TITLE
Move Azure build pipelines to OneBranch

### PR DESCRIPTION
**Pull request by getrou**
As part of compliance with updated security requirements, production pipelines must run in a 1ES environment. To do this, pipelines are converting to use OneBranch pipeline templates.

This changes how artifacts are published and the build environment is slightly different, resulting in some small changes to the build files.

As a backup plan, the existing azure pipeline scripts are left in place. These will be removed in the future.

### CODE REVIEW

File: AzurePipelinesTemplates/win32metadata-onebranch.yml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      parameters:
+        RepoDirectory: ${{ parameters.RepoDirectory }}
+
+    - task: UseDotNet@2
+      displayName: ⚙ Install .NET SDK
+      inputs:
+        packageType: sdk
+        useGlobalJson: true
+
+    # TODO - this no longer sets an Azure Devops pipeline build number, OneBranch has a different way of doing that
```
Comment (mikebattista) [inline AzurePipelinesTemplates/win32metadata-onebranch.yml:46]: Is this TODO still pending?

Comment (getrou) [reply AzurePipelinesTemplates/win32metadata-onebranch.yml:46]: It is, this Azure Devops pipeline name doesn't get properly set right now

Comment (mikebattista) [reply AzurePipelinesTemplates/win32metadata-onebranch.yml:46]: The current release pipeline uses the pipeline build number so I don't think we can merge this until this is resolved.

**Comment by riverar** [comment]
Does it make sense to migrate the pipeline to GitHub Actions and call out to a backend for signing?

**Comment by mikebattista** [comment]
No. It looks like we still have access to the existing pipelines, though, so I plan to push out a release this week.

**Comment by AArnott** [comment]
@georou What is your plan for a release pipeline? I don't see one introduced by this PR.

FWIW, for all the 1ES PT pipeline migrations I've done, I've just reused the existing YAML files for the 1ES PT variety, which kept me from having to create new pipelines in AzDO. It also avoided having YAML files for deprecated pipelines that could confuse folks who want to look at the new pipeline files.

**Comment by riverar** [comment]
Still unclear to me why this isn't just a normal GHA workflow.

**Comment by AArnott** [comment]
@riverar put simply, Microsoft has compliance requirements for shipping software that to date is only implemented on Azure Pipelines, AFAIK.

**Comment by riverar** [comment]
Are we shipping _software_ in this repository though? It's a bunch of non-executable metadata. 😂 (I understand Microsoft is sensitive to this right now though!)

**Comment by mikebattista** [comment]
Yes we ship binary tools in addition to metadata and NuGet packages that are signed by Microsoft. GHA doesn't support our workflows.

**Comment by getrou** [comment]
> @georou What is your plan for a release pipeline? I don't see one introduced by this PR.
> 
> FWIW, for all the 1ES PT pipeline migrations I've done, I've just reused the existing YAML files for the 1ES PT variety, which kept me from having to create new pipelines in AzDO. It also avoided having YAML files for deprecated pipelines that could confuse folks who want to look at the new pipeline files.

All these pipelines now run in the github-private/microsoft ADO project, which has its own (very slim) repo associated with it, and uses this github repository as a pipeline resource. The release pipeline YAML has moved there, but we could move it here if we want, I think.

**Pull request merged**